src/components/new-safe/create/logic/utils.test.ts

```diff
@@ -0,0 +1,44 @@
+import * as creationUtils from '@/components/new-safe/create/logic/index'
+import { getAvailableSaltNonce } from '@/components/new-safe/create/logic/utils'
+import * as web3Utils from '@/hooks/wallets/web3'
+import { faker } from '@faker-js/faker'
+import { BrowserProvider, type Eip1193Provider } from 'ethers'
+
+describe('getAvailableSaltNonce', () => {
+  jest.spyOn(creationUtils, 'computeNewSafeAddress').mockReturnValue(Promise.resolve(faker.finance.ethereumAddress()))
+
+  const mockProvider = new BrowserProvider(jest.fn() as unknown as Eip1193Provider)
+  const mockDeployProps = {
+    safeAccountConfig: {
+      threshold: 1,
+      owners: [faker.finance.ethereumAddress()],
+      fallbackHandler: faker.finance.ethereumAddress(),
+    },
+  }
```

Comment:
Should we put this in a `beforeAll()` block?

Reply:
From what I found online, if the data is synchronous its ok to leave it in the `describe` block but otherwise it should be in `beforeAll`. Only downside of using `beforeAll` is that we have to initialize the variables beforehand and add explicit types i.e. `let mockProvider: BrowserProvider` etc.

Reply:
Okay. Let's leave it as is then :)